### PR TITLE
re-align RFC with implementation as part of comit-rs/#1187

### PR DESCRIPTION
**Pull request by da-kami**
* define decline reasons
* remove details from decline reasons (might be re-added at later point)

### CODE REVIEW

File: RFC-002-SWAP.md [hunk 1]
```diff
@@ -235,32 +235,26 @@ Each heading represents a `reason`.
...
+#### `unsupported-swap`
 
-A ledger referenced by the sending party is unknown to the receiving party.
-Note that different networks of the same blockchain are different ledgers!
-Bitcoin Testnet is a different ledger than Bitcoin Mainnet.
+The ledger-asset combination specified in the SWAP request is not supported by the receiving party.
 
-##### `details`
+This can mean that:
+1. The ledger is not known
```
Comment (bonomat) [inline RFC-002-SWAP.md]:
```suggestion
1. The ledger is unknown
```

File: RFC-002-SWAP.md [hunk 2]
```diff
@@ -235,32 +235,26 @@ Each heading represents a `reason`.
...
+2. The asset is not known
```
Comment (bonomat) [inline RFC-002-SWAP.md]:
```suggestion
2. The asset is unknown
```

File: RFC-002-SWAP.md [hunk 3]
```diff
@@ -235,32 +235,26 @@ Each heading represents a `reason`.
...
+3. Ledger and asset are not known
```
Comment (bonomat) [inline RFC-002-SWAP.md]:
```suggestion
3. Ledger and asset are unknown
```

File: RFC-002-SWAP.md [hunk 4]
```diff
@@ -235,32 +235,26 @@ Each heading represents a `reason`.
...
+The error cases for these variants have to be properly defined first.
 
-#### `unknown-asset`
+#### `missing-header`
 
-An asset referenced by the sending party is unknown to the receiving party.
+A mandatory header expected by the receiving party is not properly specified by the sending party.
 
-##### `details`
+#### `malformed-json`
```
Comment (thomaseizinger) [inline RFC-002-SWAP.md]: As mentioned offline, this should probably be removed.

Comment (da-kami) [reply RFC-002-SWAP.md]: was renamed to `unexpected-json-field` - I don't see a reason to remove it if it is a known error. If it should really not be part of the protocol I can remove it though.

Comment (thomaseizinger) [reply RFC-002-SWAP.md]: > was renamed to `unexpected-json-field` - I don't see a reason to remove it if it is a known error. If it should really not be part of the protocol I can remove it though.

I would argue that it should not be. The fact that this error can occur is implementation specific. As mentioned in our previous discussions, the body has to be optional anyway so I would just decline with an empty body in our implementation.
Error messages are really only useful to the counterparty if they can change their behaviour and hence resolve the problem. Internal failures and bugs don't fall into that category and IMO should not be part of the protocol.

Comment (da-kami) [reply RFC-002-SWAP.md]: It is only application specific in the way, that it relates to `JSON`, but that is specified in the RFC at the moment. 

It is a deserialisation error for a faulty request that came from the other party. So for me it does make sense to inform the other party about the problem. Arguably there could be details (i.e. which field exactly cause the error), but that can be added later. 

Why would you classify this as internal failure/bug?

Comment (thomaseizinger) [reply RFC-002-SWAP.md]: > It is a deserialisation error for a faulty request that came from the other party. So for me it does make sense to inform the other party about the problem. Arguably there could be details (i.e. which field exactly cause the error), but that can be added later.

Fine with me, we probably need to revise this anyway. Let's roll with it :)

**Pull request merged**